American Whitewater/Helpers/AWApiReachHelper.swift

```diff
@@ -0,0 +1,620 @@
+                    print("HTTP Response: \(response.response.debugDescription)")
+
+                    callbackError(error)
+            }
+            
+        }
+        
+    }
+    
+    
+    func findOrNewReach(newReach: AWReach, context: NSManagedObjectContext) -> Reach {
+        let request = Reach.fetchRequest() as NSFetchRequest<Reach>
+        //print("nReach name: \(newReach.name ?? "na") ID: \(NSNumber(value: newReach.id ?? 0))")
+        guard let id = newReach.id else {
+            print("invalid id: \(newReach.id ?? -1)")
+            return Reach(context: context)            
+        }
+        
+        request.predicate = NSPredicate(format: "id == %i", id)
+                
+        guard let result = try? context.fetch(request), result.count > 0 else {
+            let reach = Reach(context: context)
+            reach.id = Int16(newReach.id ?? 0)
+            return reach
+        }
+
+        return result.first!
+    }
+
+    private func createOrUpdateReaches(newReaches: [AWReach], context: NSManagedObjectContext) {
+        context.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy
+        
+        for newReach in newReaches {
+
+            let reach = findOrNewReach(newReach: newReach, context: context)
+
+            reach.name = newReach.name
+            reach.sortName = newReach.section
+            reach.putInLat = newReach.plat
+            reach.putInLon = newReach.plon
+            reach.takeOutLat = newReach.tlat
+            reach.takeOutLon = newReach.tlon
+            reach.currentGageReading = newReach.gauge_reading // ** check
+            reach.lastGageReading = newReach.last_gauge_reading
+            reach.id = Int16(newReach.id ?? 0)
+            reach.difficulty = newReach.classRating
+            reach.condition = newReach.cond
+            reach.unit = newReach.unit
+            reach.rc = newReach.rc
+            reach.delta = newReach.reading_delta
+            reach.gageId = Int32(newReach.gauge_id ?? -1)
+            reach.gageMetric = Int16(newReach.gauge_metric ?? -1)
+            reach.gageMax = newReach.gauge_max
+            reach.gageMin = newReach.gauge_min
+            reach.state = newReach.state
+            
+            if let altName = newReach.altname, !altName.isEmpty {
+                reach.section = altName
+            } else {
+               reach.section = newReach.section
+            }
+
+            // API returns the total seconds before the present time as a string
+            // converting this to a Date
+            if let updatedSecondsString = newReach.last_gauge_updated,
+                let updatedSecondsAgo = Int(updatedSecondsString) {
+                reach.gageUpdated = Date().addingTimeInterval(TimeInterval(-updatedSecondsAgo))
+            }
+            
+            // calcualte the distance from the user
+            if let distance = newReach.distanceFrom(location:
```

Comment:
Shouldn't this be calculated when needed? Otherwise we're potentially showing stale data and/or unnecessarily performing this calculation. Distance from user isn't really a part of the reach datatype.